src/mpid/ch4/shm/posix/posix_win.h

```diff
@@ -507,7 +507,7 @@ MPL_STATIC_INLINE_PREFIX int MPIDI_POSIX_mpi_win_free_hook(MPIR_Win * win)
 
         mpi_errno = MPIDI_CH4U_destroy_shm_segment(sizeof(MPL_proc_mutex_t),
                                                    &posix_win->shm_mutex_segment_handle,
-                                                   &posix_win->shm_mutex_ptr);
+                                                   (char **) &posix_win->shm_mutex_ptr);
```

Comment:
Does it really have to be `char **`? If pointer arithmetic is not involved, I would prefer `void **` for this kind of types.

Reply:
Particularly since takes a `void **` anyway.